src/kde_analysis/eval_m1m2chieff_poiss_weights.py

```diff
@@ -187,98 +437,170 @@ def get_rate_m_chieff2D(m1_query, m2_query, Rate):
     group = hdf[iter_name]
     if 'bootstrap_weights' in group:
         boots_weighted = True
-        poisson_weights = group['bootstrap_weights'][:]
-        assert min(poisson_weights) > 0, "Some bootstrap weights are non-positive!"
-        Nboots = poisson_weights.sum()  # Number of events in bootstrap
+        weights = group['bootstrap_weights'][:]
+        # Allow zero weights as corner case
+#       assert min(weights) > 0, "Some bootstrap weights are non-positive!"
+        Nboots = weights.sum()  # Number of events in bootstrap
 
     # Check if VT weighting should be used
     if 'rwvt_vals' in group:
         vt_weights = True
         vt_vals = group['rwvt_vals'][:] / 1e9  # change units to Gpc^3
 
     samples = group['rwsamples'][:]
+    if boots_weighted:
+        # Remove samples with zero bootstrap weight as they may have bad behaviour in
+        # adaptive KDE (due to extremely small pilot density at the sample location)
+        # and have no effect on the KDE, and to reduce compute cost
+        samples = samples[weights > 0., :]
+        if vt_weights: vt_vals = vt_vals[weights > 0.]
+        # Make sure all arrays are the same length
+        weights = weights[weights > 0.]
+
     alpha = group['alpha'][()]
     bwx = group['bwx'][()]
     bwy = group['bwy'][()]
     bwz = group['bwz'][()]
-
+    # Check symmetric dimensions: bwx vs bwy for m1-m2 symmetry
+    if not np.isclose(bwx, bwy, rtol=1e-8, atol=0.0):
+        print(f"WARNING: bwx != bwy (bwx={bwx}, bwy={bwy}). Setting both to geometric mean.")
+        gm = float(np.sqrt(bwx * bwy))  # valid for [0, 1]
```

Comment:
OK, but I'm not following why we need to use `float` and what the comment means.  The calculation should be valid for any values >0